Adding support for google oauth2 authentication in the omni-executor

silva-fj opened the pull request:
This PR introduces support for google oauth2 authentication for native calls as well as some refactoring: Renamed the `primitives` crate to `executor-primitives` and updated all relevant references in the codebase.

silva-fj committed:
adding oauth2_state_verifier storage

silva-fj committed:
initial setup google oauth2 identity verification

silva-fj committed:
setting up google's IdToken claims and decode utils

silva-fj committed:
exporting AccountId from heima-primitives

silva-fj committed:
small refactoring

silva-fj committed:
fix: add Display impl for AuthenticationError

Implements Display trait for AuthenticationError to provide better error
messages for logging and user feedback. This also fixes the dead code
warning for AuthTokenError variant.

silva-fj committed:
feat: add Google OAuth2 client configuration

silva-fj committed:
feat: implement Google OAuth2 verification

Adds implementation of Google OAuth2 token verification for
authentication. This includes:
- State verifier validation
- Token exchange using authorization code
- ID token decoding and validation
- Identity verification against sender's identity

silva-fj committed:
fix: use sender identity for auth token claims

Changes auth token claims to use sender identity hash instead of omni
account. This simplifies the auth token verification process by removing
the dependency on omni account storage and making the verification more
direct.

silva-fj committed:
refactor: rename primitives to executor-primitives

silva-fj committed:
refactor: clean up imports and unused code

silva-fj committed:
fix: rename oauth state storage key

silva-fj committed:
refactor: remove unused executor-crypto dependency

kziemianek commented on tee-worker/omni-executor/heima/identity-verification/src/web2/google/mod.rs:
```diff
@@ -0,0 +1,102 @@
+mod oauth2_client;
```
Can this be moved out of identity-verificaiton / heima context ? To me it looks like just general oauth

silva-fj replied:
Sure, in the identity worker we have this in `litentry/core/data-providers` but since we are not sure if we are going to support the identity/vc related stuff in the omni-executor I didn't create that structure. I'll create an `oauth-providers` crate as we might add support for others

silva-fj committed:
feat: add Google OAuth2 authorization URL endpoint

Implements a new RPC endpoint `omni_getOAuth2GoogleAuthorizationUrl` that
generates and returns an authorization URL for Google OAuth2 authentication
flow. The endpoint:

- Takes Google account and redirect URI as parameters
- Stores state verifier in database for security
- Returns hex-encoded authorization URL
- Integrates with existing identity verification system

silva-fj committed:
feat: add oauth-providers crate for Google OAuth2

silva-fj committed:
refactor: move GoogleOAuth2Client to oauth-providers

Moves Google OAuth2 client implementation from identity-verification crate
to the new oauth-providers crate for better code organization.

silva-fj committed:
Merge branch 'dev' into p-1284-add-support-for-google-oauth2-authentication
